storm-client/src/jvm/org/apache/storm/windowing/TimeEvictionPolicy.java

```diff
@@ -54,7 +62,22 @@ public void track(Event<T> event) {
 
     @Override
     public void setContext(EvictionContext context) {
-        this.evictionContext = context;
+        EvictionContext prevContext = evictionContext;
+        evictionContext = context;
+        // compute window length adjustment (delta) to account for time drift
+        if (context != null && context.getSlidingInterval() != null) {
```

Comment:
Nitpick: I'd rather not check for null unless there's a good reason. As far as I can tell there's no reason to reset the EvictionContext to null, so I'd rather we get an NPE if it happens unexpectedly, so we can fix what is likely a bug.